test/cluster/mzcompose.py

```diff
@@ -2929,7 +2929,19 @@ def validate_since(since: int, name: str) -> None:
     ), f"index sinces did not match {index_since1} vs {index_since2})"
 
 
-RE_WORKLOAD_CLASS_LABEL = re.compile(r'workload_class="(?P<value>\w+)"')
+def find_proxy_port(logs: str, cluster_id: str, port_name: str) -> int | None:
+    """Extract a proxy port from the given logs."""
+
+    RE_PROXY_LINE = re.compile(
+        rf".*INFO mz_orchestrator_process: cluster-{cluster_id}-replica-.*:"
+        rf" {port_name} tcp proxy listening on 0.0.0.0:(?P<port>\d+)"
+    )
+
+    for line in logs.splitlines():
+        if match := RE_PROXY_LINE.match(line):
+            return int(match.group("port"))
+
+    return None
 
 
 def workflow_test_workload_class_in_metrics(c: Composition) -> None:
```

Comment:
The test previously didn't catch this because it used an unmanaged cluster to easily get at the clusterd metrics, but applying the workload class was only broken with managed clusters. Fixed by using a managed cluster and getting the internal-http port from the envd logs.